benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/DistributedWorkersEnsemble.java

```diff
@@ -245,54 +235,31 @@ public PeriodStats getPeriodStats() {
-            } catch (Exception e) {
-                log.error("Failed to decode publish latency");
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.scheduleLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                    ByteBuffer.wrap(is.scheduleLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode schedule latency");
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.publishDelayLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.publishDelayLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode publish delay latency: {}",
-                          ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.publishDelayLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.publishDelayLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.publishDelayLatencyBytes), TimeUnit.SECONDS.toMicros(30)));
-            } catch (Exception e) {
-                log.error("Failed to decode publish delay latency: {}",
-                          ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.publishDelayLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-
-            try {
-                stats.endToEndLatency.add(Histogram.decodeFromCompressedByteBuffer(
-                        ByteBuffer.wrap(is.endToEndLatencyBytes), TimeUnit.HOURS.toMicros(12)));
-            } catch (Exception e) {
-                log.error("Failed to decode end-to-end latency: {}",
-                        ByteBufUtil.prettyHexDump(Unpooled.wrappedBuffer(is.endToEndLatencyBytes)));
-                throw new RuntimeException(e);
-            }
-        });
+        final CumulativeLatencies stats = new CumulativeLatencies();
+        individualStats.forEach(is -> Map.of(
+                        stats.publishLatency, is.publishLatencyBytes,
+                        stats.scheduleLatency, is.scheduleLatencyBytes,
+                        stats.publishDelayLatency, is.publishDelayLatencyBytes,
+                        stats.endToEndLatency, is.endToEndLatencyBytes)
+                .forEach((histogram, bytes) -> {
+                    final ByteBuffer buffer = ByteBuffer.wrap(bytes);
+                    try {
+                        histogram.add(Histogram.decodeFromCompressedByteBuffer(buffer,
+                                TimeUnit.SECONDS.toMicros(30)));
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        log.error("Error adding to histogram {}: {}", histogram, e);
+                        throw new RuntimeException(e);
+                    } catch (DataFormatException e) {
```

Comment:
I think you could just use the existing style of catching exception, but only around the `Histogram.decodeFromCompressedByteBuffer` since that's the only place where we want to do the histo output?

I don't think we need to catch AIOOBE here at all, it will be logged nicely at the higher level when we fail. So we onnly need special handling for the histo since we want to print it out.